framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -923,6 +1066,50 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
                 "ValueAnnotatedTypeFactory: unexpected class: " + first.getClass());
     }
 
+    public AnnotationMirror createIntRangeAnnotation(Range range) {
```

Comment:
I feel that this should take a well-formed Range as its argument.  (That range might be Range.NOTHING, or Range.EVERYTHING, but in those cases use an abstraction such as a query method or comparison against a constant, rather than checking fields directly.)  Ill-formed Range objects should not exist.